vpr/src/route/rr_graph_uxsdcxx_serializer.h

```diff
@@ -1626,39 +1631,139 @@ class RrGraphSerializer final : public uxsd::RrGraphBase<RrGraphContextTypes> {
-                return BOTTOM;
+                side_mask.set(BOTTOM);
+                break;
+            case uxsd::enum_loc_side::BOTTOM_LEFT:
+                side_mask.set(BOTTOM);
+                side_mask.set(LEFT);
+                break;
+            case uxsd::enum_loc_side::LEFT:
+                side_mask.set(LEFT);
+                break;
+            case uxsd::enum_loc_side::TOP_RIGHT_BOTTOM_LEFT:
+                side_mask.set(TOP);
+                side_mask.set(RIGHT);
+                side_mask.set(BOTTOM);
+                side_mask.set(LEFT);
                 break;
             default:
                 report_error(
                     "Invalid side %d", side);
         }
-    }
-
-    uxsd::enum_loc_side to_uxsd_loc_side(e_side side) {
-        switch (side) {
-            case LEFT:
-                return uxsd::enum_loc_side::LEFT;
-            case RIGHT:
-                return uxsd::enum_loc_side::RIGHT;
-            case TOP:
-                return uxsd::enum_loc_side::TOP;
-            case BOTTOM:
-                return uxsd::enum_loc_side::BOTTOM;
-            default:
-                report_error(
-                    "Invalid side %d", side);
+        return side_mask;
+    }
+
+    /* The enumeration MUST follow the order of side maskcode
+     * so that it is easy for RRGraph writer to convert the mask code to enumerations 
+     *
+     * index | LEFT BOTTOM RIGHT TOP | mask
+     * ======+======================+======
+     *  1    |                    X  | 0001
+     * ======+======================+======
+     *  2    |               X       | 0010
+     * ======+======================+======
+     *  3    |               X    X  | 0011
+     * ======+======================+======
+     *  4    |         X             | 0100
+     * ======+======================+======
+     *  5    |         X          X  | 0101
+     * ======+======================+======
+     *  6    |         X     X       | 0110
+     * ======+======================+======
+     *  7    |         X     X    X  | 0111
+     * ======+======================+======
+     *  8    |  X                    | 1000
+     * ======+======================+======
+     *  9    |  X                 X  | 1001
+     * ======+======================+======
+     *  10   |  X            X       | 1010
+     * ======+======================+======
+     *  11   |  X            X    X  | 1011
+     * ======+======================+======
+     *  12   |  X      X             | 1100
+     * ======+======================+======
+     *  13   |  X      X          X  | 1101
+     * ======+======================+======
+     *  14   |  X      X     X       | 1110
+     * ======+======================+======
+     *  15   |  X      X     X    X  | 1111
+     */
+    uxsd::enum_loc_side to_uxsd_loc_side(std::bitset<NUM_SIDES> sides) {
+        std::array<uxsd::enum_loc_side, 16> side_map = {uxsd::enum_loc_side::UXSD_INVALID,
```

Comment:
Can I make a suggestion on how to construct this?

Maybe in the constructor have something like:
```
std::array<uxsd::enum_loc_side, 16> side_map;
side_map[0] = uxsd::enum_loc_side::UXSD_INVALID;
side_map[(1 << TOP)] = uxsd::enum_loc_side::TOP;
side_map[(1 << RIGHT)] = uxsd::enum_loc_side::RIGHT;
side_map[(1 << TOP) | (1 << RIGHT)] = uxsd::enum_loc_side::TOP_RIGHT;
...
```

Reply:
Make senses. It is more straightforward to read and maintain.